### PR TITLE
scan: patch cosmosjs binding to support sending MsgRequest transaction

### LINKED ISSUE #1038: Patch cosmosjs binding to support sending MsgRequest transaction
Can consult with Swit on design and abstraction, but take further work from what @evilpeach did and make the library generic enough for sending request transaction.

### PR DESCRIPTION
**Pull request by tansawit**
Fixes: #1038

### CODE REVIEW

File: scan/src/bindings/StdMsgRequest.re [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  type_: string,
+  value: send_value_t,
+};
+
+type fee_t = {
+  amount: array(amount_t),
+  gas: string,
+};
+
+type t = {
```
Comment (sorawit) [inline scan/src/bindings/StdMsgRequest.re:28]: Let's create a wrapper function that helps creating `t` from known types (`Address.t`, `int`, etc) so the caller is safe.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/bindings/StdMsgRequest.re [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  msgs: array(msg_t),
+  chain_id: string,
+  fee: fee_t,
+  memo: string,
+  account_number: string,
+  sequence: string,
+};
+
+let create_t =
```
Comment (sorawit) [inline scan/src/bindings/StdMsgRequest.re]: Use camelcase for function/variable names.

File: scan/src/bindings/StdMsgRequest.re [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      feeAmount,
+      accountNumber,
+      sequence,
+    ) => {
+  let returnT: t = {
+    msgs: [|
+      {
+        type_: "zoracle/Request",
+        value: {
+          oracleScriptID: string_of_int(oracleScriptID),
```
Comment (sorawit) [inline scan/src/bindings/StdMsgRequest.re]: Stick to one pattern: `string_of_int(x)` or `x |> string_of_int`

File: scan/src/bindings/StdMsgSend.re [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+type t = {
+  msgs: array(msg_t),
+  chain_id: string,
+  fee: fee_t,
+  memo: string,
+  account_number: string,
+  sequence: string,
+};
+
+let create_t = (fromAddress, toAddress, sendAmount, accountNumber, sequence) => {
```
Comment (sorawit) [inline scan/src/bindings/StdMsgSend.re]: ditto here.

File: scan/src/bindings/StdMsgRequest.re [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    chain_id: "bandchain",
+    fee: {
+      amount: [|{amount: feeAmount, denom: "uband"}|],
+      gas: 3000000 |> string_of_int,
+    },
+    memo: "",
+    account_number: accountNumber,
+    sequence,
+  };
+  returnT;
```
Comment (sorawit) [inline scan/src/bindings/StdMsgRequest.re]: Can't you use return without creating a temporary variable?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/bindings/StdMsgRequest.re [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+type t = {
+  msgs: array(msg_t),
+  chain_id: string,
+  fee: fee_t,
+  memo: string,
+  account_number: string,
+  sequence: string,
+};
+
+let createT =
```
Comment (sorawit) [inline scan/src/bindings/StdMsgRequest.re]: Can't we just name it `create`? What's that `T` for?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/bindings/StdMsgSend.re [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+type t = {
+  msgs: array(msg_t),
+  chain_id: string,
+  fee: fee_t,
+  memo: string,
+  account_number: string,
+  sequence: string,
+};
+
+let createT = (fromAddress, toAddress, sendAmount, accountNumber, sequence) => {
```
Comment (taobun) [inline scan/src/bindings/StdMsgSend.re]: Just name it as `create`

File: scan/src/bindings/StdMsgRequest.re [hunk 8]
```diff
@@ -0,0 +1,73 @@
+type amount_t = {
+  amount: string,
+  denom: string,
+};
+
+type send_value_t = {
```
Comment (taobun) [inline scan/src/bindings/StdMsgRequest.re]: request_value_t

File: scan/src/bindings/StdMsgRequest.re [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+  {
+    msgs: [|
+      {
+        type_: "zoracle/Request",
+        value: {
+          oracleScriptID: oracleScriptID |> string_of_int,
+          calldata: calldata |> JsBuffer.toBase64,
+          requestedValidatorCount: requestedValidatorCount |> string_of_int,
+          sufficientValidatorCount: sufficientValidatorCount |> string_of_int,
+          expiration: 20 |> string_of_int,
```
Comment (taobun) [inline scan/src/bindings/StdMsgRequest.re]: Can we make it as optional parameter? (~expiration=20)

File: scan/src/bindings/StdMsgRequest.re [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+          prepareGas: 20000 |> string_of_int,
+          executeGas: 150000 |> string_of_int,
```
Comment (taobun) [inline scan/src/bindings/StdMsgRequest.re]: ditto

File: scan/src/bindings/StdMsgRequest.re [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+          sender: sender |> Address.toBech32,
+        },
+      },
+    |],
+    chain_id: "bandchain",
+    fee: {
+      amount: [|{amount: feeAmount, denom: "uband"}|],
+      gas: 3000000 |> string_of_int,
```
Comment (taobun) [inline scan/src/bindings/StdMsgRequest.re]: gas too

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by prin-r — DISMISSED** [review]

**Review by sorawit — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]
LGTM

**Review by taobun — APPROVED** [review]

**Pull request merged**